cEP-0021.md

```diff
@@ -0,0 +1,266 @@
+# Debug and Profile Bears
+
+| Metadata |                                                   |
+| -------- | ------------------------------------------------- |
+| cEP      | 0019                                              |
+| Version  | 1.0                                               |
+| Title    | Debug and Profile Bears                           |
+| Authors  | Vaibhav Kumar Rai <mailto:raivaibhav08@gmail.com> |
+| Status   | Proposed                                          |
+| Type     | Process                                           |
+
+## Abstract
+
+This cEP describes the detailed implementation of profiling interface as a part
+of the
+[GSoC'18 project](https://summerofcode.withgoogle.com/projects/#6109762077327360).
+
+## Introduction
+
+Profiling a Python program is doing a dynamic analysis that measures the
+execution time of the program and everything that compose it. That means
+measuring the time spent in each of its functions. This will give the data about
+where our program is spending time, and what area might be worth optimizing,
+thus we can provide a *Profiling interface* from within coala and make it
+available to bear writers so that they can create Better Performant bears.
+
+This proposal is to introduce the Profiling interface so that bear writer will
+have the ability to jump directly into Bear code and move through it using a
+debugger interface and Profile the Bear code to create Better Performant
+bears.
+
+## Proposed Change
+
+Bear writer can invoke Profiler through *CLI*:
+
+- By passing custom settings through *CLI*
+  `coala -b Bear Mybear --files file --profile-bears -"True,...."`,
+- By using coala default settings provided by coala
+  `coala -b Bear Mybear --files file --profile-bears`,the default are to strip
+  the directory (`strip_dirs()`) and sort the stats of profiler result by
+  cumulative time (`cumtime`) and display the profiler result on *CLI*
+
+Bear writer can also invoke the Profiler by using `coafile` i.e.
+`profile_bears = XXX`.
```

Comment:
please explain `xxx` and maybe put a more descriptive placeholder there.

is this supposed to be a list of bear names to profile? why's this setting different then the CLI ones?

since you can also specify the bears that are supposed to be run by coala with the `-b` flag like in your example above, a generic `profile-bears` setting might actually be more appropriate here. But that's up for debate, what are your thoughts?

Reply:
Actually I want to give user a way to sort the profiler ouput by their own

Reply:
>why's this setting different then the CLI ones?

It's not different